### PR TITLE
Release v0.6.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v0.6.0

## Release details

Autonomy:
- Removes the need for the intermediate agent layer when scaffolding an `FSM` app using specification
- Updates the tendermint communication server
  - `GET /params` endpoint to return `peer_id` for the local tendermint node
  - `POST /params` endpoint to update `persistent_peers` when updating the chain config

Packages:
- Fixes the `fast_forward_to_behaviour` in the `abci` skill test tools
- Adds support for auto generated behaviour IDs
- Adds support for auto-generated round IDs
- Introduces the `get_name` method as part of the `abstract_round_abci`skill to retrieve the name of a property dynamically
- Updates the various packages to to newly introduced auto generate functionalities
- Removes sleep on `reset_tendermint_with_wait` on startup
- Updates the `a2a` transaction logic to wait for block production to begin before sending an `a2a` transaction via tendermint
- Updates the `RegistrationStartupBehaviour` and `TendermintHandler` to make sure we update the persistent peers when establishing a new chain
- Adds support for specifying the external host name for the tendermint P2P connection

Tests
- Adds tests for `termination_abci`
- Adds tests for docstring analyser when no `AbciApp` definition is found in the provided module
- Adds test coverage for `autonomy develop` command group
- Introduces pre and post conditions checks on the `SynchronizedData` for each initial and final state

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side), from `develop`
- [x] I've updated the dependencies versions to the latest, wherever is possible.
- [x] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [x] I built the documentation and updated it with the latest changes
- [x] I've added an item in `HISTORY.md` for this release
- [x] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
 # Release History - `open-autonomy`
 
 
+# 0.6.0 (2022-12-16)
+
+Autonomy:
+- Removes the need for the intermediate agent layer when scaffolding an `FSM` app using specification
```
Comment (angrybayblade) [inline HISTORY.md:7]: @jmoreira-valory Please create a ticket for the next sprint to update the tutorials accordingly

Comment (DavidMinarsch) [reply HISTORY.md:7]: No idea what this update refers to.

File: HISTORY.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+- Updates the tendermint communication server
+  - `GET /params` endpoint to return `peer_id` for the local tendermint node
+  - `POST /params` endpoint to update `persistent_peers` when updating the chain config
+
+Packages:
+- Fixes the `fast_forward_to_behaviour` in the `abci` skill test tools
```
Comment (DavidMinarsch) [inline HISTORY.md:13]: Fixes in what way? Missing half a sentence like `... by doing X`

File: HISTORY.md [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+- Adds support for auto generated behaviour IDs
+- Adds support for auto-generated round IDs
+- Introduces the `get_name` method as part of the `abstract_round_abci`skill to retrieve the name of a property dynamically
+- Updates the various packages to to newly introduced auto generate functionalities
+- Removes sleep on `reset_tendermint_with_wait` on startup
+- Updates the `a2a` transaction logic to wait for block production to begin before sending an `a2a` transaction via tendermint
+- Updates the `RegistrationStartupBehaviour` and `TendermintHandler` to make sure we update the persistent peers when establishing a new chain
+- Adds support for specifying the external host name for the tendermint P2P connection
```
Comment (DavidMinarsch) [inline HISTORY.md:21]: Also:
`- Introduces pre- and post-conditions on the synchronized data for each initial and final state of an FSM app (including default pre conditions). These are verified during chaining of the apps.`
`- Updates all FSM apps to specify their pre- and post-conditions.`

File: HISTORY.md [hunk 4]
```diff
@@ -1,6 +1,32 @@
...
+
+Tests
+- Adds tests for `termination_abci`
+- Adds tests for docstring analyser when no `AbciApp` definition is found in the provided module
+- Adds test coverage for `autonomy develop` command group
+- Introduces pre and post conditions checks on the `SynchronizedData` for each initial and final state
```
Comment (DavidMinarsch) [inline HISTORY.md:27]: Merge with above - it's not about tests

File: docs/upgrading.md [hunk 5]
```diff
@@ -5,6 +5,13 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open Autonomy
 
+## `v0.5.0.post2` to `v0.6.0`
+
+Breaking changes
+
+- The pre and post conditions needs to be defined while the FSM
```
Comment (DavidMinarsch) [inline docs/upgrading.md:12]:
```suggestion
- Optional: The developer can update to use the auto-generated behaviour and round ids. In this case, the `behaviour_id` and `round_id` need no longer be set on the class. When accessing the auto generated ids on the class use `auto_behaviour_id()` and `auto_round_id()` class methods. On the instance `behaviour_id` and `round_id` properties resolve the auto id class methods, so no change is required.
- Required: The pre- (`db_pre_conditions`) and post- (`db_post_conditions`) conditions on the synchronized data need to be defined for each initial and final state in an `AbciApp`. See `transaction_settlement_abci` for an example.
```

Comment (DavidMinarsch) [reply docs/upgrading.md:12]: @Adamantios suggest you add a couple of lines in the docs on the pre and post conditions and the default conditions

File: packages/packages.json [hunk 6]
```diff
@@ -1,38 +1,38 @@
...
+        "agent/valory/registration_start_up/0.1.0": "bafybeief2p5ngx5siudg55bjq3z6iuvtaojlnegqwkihgodtiywdfueiy4",
+        "agent/valory/test_abci/0.1.0": "bafybeig2s5uiomakecuzcfm76icvkqr7xdmywm4h54oildppj73xe2oe24",
+        "service/valory/counter/0.1.0": "bafybeihi6l2zaqbacykstg45qftonkvmuxrwbevxh4emrvwl4qsnqi6xhi",
+        "service/valory/hello_world/0.1.0": "bafybeidkly5wgeopyxom5bwhtrq57ckfei7x6zavi4bowaranijutpefde",
+        "service/valory/register_reset/0.1.0": "bafybeiaxkrt7c47nqretkhw6mexqkbxibbdjn7z6vooxqrzcui2witrnqy",
+        "skill/valory/safe_deployment_abci/0.1.0": "bafybeidfueovwewv47g2juk6dzgit3rvzxur34gqqb2ghq56vjfv56h7ji",
+        "skill/valory/register_reset_recovery_abci/0.1.0": "bafybeihfzm2czimnq5nmotaxr2rvciuzb6smnjzom5thsdiftzazqskena",
+        "agent/valory/register_reset_recovery/0.1.0": "bafybeicdo6drxrgptldfg3vu6hgarrzeybfz2yfapb2oehxzyvipofhrge"
     },
     "third_party": {
```
Comment (DavidMinarsch) [inline packages/packages.json:37]: Please confirm we are up-to-date wrt to these. I had the same question on the PR re 1.26 of open-aea

Comment (angrybayblade) [reply packages/packages.json:37]: We haven't had a change in hashes on open-aea for past month

File: HISTORY.md (hunk 1 above)
Comment (jmoreira-valory) [reply HISTORY.md:7]: Currently (well, previously) we needed an intermediate agent to scaffold an FSM App with the scaffold tool.
Apparently this requirement has been eliminated now.

Comment (jmoreira-valory) [reply HISTORY.md:7]: A ticket has been created in Trello.

File: HISTORY.md [hunk 7]
```diff
@@ -1,6 +1,32 @@
...
+- Adds support for auto generated behaviour IDs
+- Adds support for auto-generated round IDs
+- Introduces the `get_name` method as part of the `abstract_round_abci`skill to retrieve the name of a property dynamically
+- Updates the various packages to to newly introduced auto generate functionalities
+- Removes sleep on `reset_tendermint_with_wait` on startup
+- Updates the `a2a` transaction logic to wait for block production to begin before sending an `a2a` transaction via tendermint
+- Updates the `RegistrationStartupBehaviour` and `TendermintHandler` to make sure we update the persistent peers when establishing a new chain
+- Adds support for specifying the external host name for the tendermint P2P connection
+
+Tests
```
Comment (0xArdi) [inline HISTORY.md:23]:
```suggestion
Tests
- Adds an e2e test to showcase hard reset being used as a recovery mechanism.
```

Comment (0xArdi) [reply HISTORY.md:23]: https://github.com/valory-xyz/open-autonomy/pull/1622

File: HISTORY.md [hunk 8]
```diff
@@ -1,6 +1,32 @@
...
+
+Autonomy:
+- Removes the need for the intermediate agent layer when scaffolding an `FSM` app using specification
+- Updates the tendermint communication server
+  - `GET /params` endpoint to return `peer_id` for the local tendermint node
+  - `POST /params` endpoint to update `persistent_peers` when updating the chain config
+
+Packages:
+- Fixes the `fast_forward_to_behaviour` in the `abci` skill test tools
+- Adds support for auto generated behaviour IDs
```
Comment (0xArdi) [inline HISTORY.md:14]:
```suggestion
- Adds support for auto generated behaviour IDs
- Added message handling support to `TmManager`.
- Reset the app hash to the app hash in the begging of the period when hard resetting for recovery.
```

Comment (0xArdi) [reply HISTORY.md:14]: https://github.com/valory-xyz/open-autonomy/pull/1636

File: docs/upgrading.md (hunk 5 above)
Comment (Adamantios) [reply docs/upgrading.md:12]: Done in ae17700d472b8c5317f959327a9ebf4de1aed314.

File: HISTORY.md (hunk 1 above)
Comment (DavidMinarsch) [reply HISTORY.md:7]: Ok then we need to fix the language. "layer" is not a term that's appropriate here

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**